### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by CedrikNikita**

### CODE REVIEW

**Comment by mradkov** [comment]
I would like https://github.com/aeternity/superhero-wallet/pull/519 to be merged first and then apply these changes on-top.

File: src/store/plugins/observables.js [hunk 1]
```diff
@@ -19,13 +20,34 @@ export default store => {
...
+            .catch(error => {
+              if (!isNotFoundError(error)) {
+                handleUnknownError(error);
+              }
+              return 0;
+            })
+            .then(result => convertToAE(result))
+        : Promise.resolve(0),
+    ),
+    multicast(new BehaviorSubject(0)),
```
Comment (davidyuk) [inline src/store/plugins/observables.js:38]: The initial value of the previous balance implementation was persisted in localStorage (as I understand, to don't show 0 at startup). Shouldn't we consider this feature as important and keep it in this implementation?

File: src/store/plugins/observables.js [hunk 2]
```diff
@@ -41,5 +63,17 @@ export default store => {
...
+    balance: balance$,
+    topBlockHeight: topBlockHeight$,
+    tokenBalance: balance$.pipe(
+      map(bal =>
+        store.state.current.token !== 0
+          ? store.state.tokens[store.state.current.token].balance
+          : bal,
+      ),
+    ),
+    balanceCurrency: balance$.pipe(map(bal => store.getters.currentCurrencyRate * bal)),
```
Comment (davidyuk) [inline src/store/plugins/observables.js:77]: The `balanceCurrency` won't be updated if `getters.currentCurrencyRate` changed. Would be more accurate to subscribe to `currentCurrencyRate` using `watchAsObservable`

File: src/store/plugins/observables.js [hunk 3]
```diff
@@ -41,5 +63,17 @@ export default store => {
...
   // eslint-disable-next-line no-param-reassign
-  store.state.observables = { notifications: notifications$ };
+  store.state.observables = {
+    notifications: notifications$,
+    balance: balance$,
+    topBlockHeight: topBlockHeight$,
+    tokenBalance: balance$.pipe(
+      map(bal =>
+        store.state.current.token !== 0
+          ? store.state.tokens[store.state.current.token].balance
```
Comment (davidyuk) [inline src/store/plugins/observables.js:73]: The balance will remain polled even if the balance of the tokens is shown.

**Review by davidyuk — CHANGES_REQUESTED** [review]
Proposing to extract obvious commits to a separate PR to merge them earlier.

File: src/store/plugins/observables.js [hunk 4]
```diff
@@ -25,7 +25,7 @@ export default store => {
     entityType,
     ...other,
     sender,
-    chainName: store.state.chainNames[sender],
+    chainName: store.state.chainNames && store.state.chainNames[sender],
```
Comment (davidyuk) [inline src/store/plugins/observables.js:28]:
```suggestion
    chainName: store.state.chainNames?.[sender],
```
this is not the same as https://github.com/aeternity/superhero-wallet/pull/562#discussion_r526400981

File: src/popup/router/pages/Send.vue [hunk 5]
```diff
@@ -178,15 +159,22 @@ export default {
...
         to: '',
+        from: '',
         hash: '',
       },
+      wording: {
+        sender: this.$t('pages.send.sender'),
+        recipient: this.$t('pages.send.recipient'),
+        to: this.$t('pages.send.to'),
+        from: this.$t('pages.send.from'),
+        hash: this.$t('pages.send.hash'),
```
Comment (davidyuk) [inline src/popup/router/pages/Send.vue:171]: This approach makes it more difficult to track is wording actually used or not

File: src/popup/router/pages/Send.vue [hunk 6]
```diff
@@ -50,23 +50,15 @@
...
+          <div
+            v-for="(value, key) in { sender: account.publicKey, recipient: form.address }"
+            :key="key"
+            class="info-group"
+          >
+            <label class="info-label">{{ wording[key] }}</label>
+            <span :data-cy="`review-${key}`" class="info-span" @click="openTxExplorer(value)">{{
+              value
+            }}</span>
           </div>
```
Comment (davidyuk) [inline src/popup/router/pages/Send.vue:62]: This looks ok but a more component-oriented approach would be to extract InfoGroup component and exactly specify fields without a cycle:
```suggestion
          <InfoGroup
            :value="account.publicKey"
            :label="$t('pages.send.sender')"
          />
          <InfoGroup
            :value="form.address"
            :label="$t('pages.send.recipient')"
          />
```
the necessity to pass `data-cy` is just crap, e2e tests shouldn't require that

**Review by davidyuk — APPROVED** [review]
Despite remarks, I think we can merge that

File: src/popup/router/pages/Send.vue (hunk 6 above)
Comment (davidyuk) [reply src/popup/router/pages/Send.vue:62]: Actually, we may apply `data-cy` on InfoGroup and to make selectors on e2e side a bit more complex.

**Review by davidyuk — APPROVED** [review]

**Pull request merged**